### ISSUE TITLE
Add model implementations

### ISSUE THREAD

**Issue by darsnack** [FluxML/Metalhead.jl#112]
Below are models that we still need to implement

- [ ] Inception v4 (https://arxiv.org/abs/1602.07261v2)
- [ ] Inception ResNet v2 (https://arxiv.org/abs/1602.07261v2)
- [ ] Xception (https://openaccess.thecvf.com/content_cvpr_2017/html/Chollet_Xception_Deep_Learning_CVPR_2017_paper.html)
- [x] ResNeXt
- [x] MobileNet v1
- [x] MobileNet v2
- [x] MLPMixer
- [ ] ViT

**Comment by darsnack** [comment]
cc @theabhirath please add the models you are planning on implementing to this list

**Comment by darsnack** [comment]
A starting point for EfficientNet can be found at https://github.com/pxl-th/EfficientNet.jl (see #106).

**Comment by theabhirath** [comment]
I'm planning on  

- [ ] [CaiT](https://arxiv.org/abs/2103.17239)
- [ ] [LeViT(https://arxiv.org/abs/2104.01136)
- [ ] [Swin](https://arxiv.org/abs/2103.14030)
- [ ] [MobileViT](https://arxiv.org/abs/2110.02178)
- [ ] [ConvNeXt](https://arxiv.org/abs/2201.03545)

at first. I also want to start on object detection and semantic segmentation models but there's a lot of helper functions related to the fields that I will have to write - if everything CV-related is planned to be in Metalhead, then I'll go ahead and start coding them up

**Comment by rutvij-25** [comment]
Hello, Can I work on inception v4 and efficient net?(Also I have already coded ESRGANs)

**Comment by DhairyaLGandhi** [comment]
There's #113 for efficient net. Would be good to port srgan etc into Metalhead as well

**Comment by Alexander-Barth** [comment]
I could provide the pretrained weights for VGG and ResNet converted from PyTorch (once some minor changes to the Metalhead models are merged so that they are equivalent to the PyTorch model).

(I would be very interested in an ESRGAN implementation. 😃)

**Comment by DhairyaLGandhi** [comment]
Also see #109 for pretrained models. It's hard to know before hand which weights would work well between pytorch and Flux, but if we have some pretrained weights we can validate, that would be welcome!

**Comment by darsnack** [comment]
> Hello, Can I work on inception v4 and efficient net?(Also I have already coded ESRGANs)

As mentioned, there is already a PR for EfficientNet, but InceptionNet v4 would be very welcome!

**Comment by darsnack** [comment]
> I could provide the pretrained weights for VGG and ResNet converted from PyTorch (once some minor changes to the Metalhead models are merged so that they are equivalent to the PyTorch model).

Yes, I think your flow will work well for both those models. Please submit PRs to MetalheadWeights when you have them!

**Comment by rutvij-25** [comment]
How do I contribute models here? I'm fairly new here

**Comment by darsnack** [comment]
The Flux [contribution guide](https://github.com/FluxML/Flux.jl/blob/master/CONTRIBUTING.md) has some info as well as links for how to get started with making your first contribution.

I'll briefly summarize the process for this repo (apologies if you already know this):
1. Fork this repo (there is a button to "fork" on the front page). This creates your own copy of the repo.
2. Clone your fork locally onto your machine.
3. Create a new branch for the model or feature that you are trying to add.
4. Edit the code to add the new feature or create a new file that implements the model. It would be useful to skim the existing model code so you get an idea of what helper functions already exist and the model building style that the repo follows.
5. Make commits to your feature branch and push them to your fork.
6. Go to "Pull Requests" on this repo and click "New Pull Request." Choose your feature branch from your fork as the source.
7. We will review your pull request and provide feedback. You can incorporate the feedback by making more commits to your feature branch. The pull request will update automatically. Together, we'll iterate the code until the design is ready to merge.
8. At this point, one of the maintainers will merge your pull request, and you're done!

**Comment by Alexander-Barth** [comment]
Indeed!, In this PR I have taken a similar approach (but calling PyTorch via PyCall):
https://github.com/FluxML/MetalheadWeights/pull/2

Sometimes the weight have to be transposed as you found by the error message.
https://github.com/FluxML/MetalheadWeights/pull/2/files#diff-796965064d4fe9651f28a44c7b3dcd003b4690f37e619bb91782c283cee5a159R99

The ability to use PyTorch weights without needing PyTorch installed is indeed very nice!

**Comment by pri1311** [comment]
I would like to give LeViT a try, if someone else is not already working on it.

**Comment by shivance** [comment]
@darsnack I would like to take a crack at MobileVIT !

**Comment by darsnack** [comment]
Go for it! For both ViT-based models, make sure to look at the existing ViT implementation and the Layers submodules. Torchvision models are good reference, but ultimately we want something that is a Flux-idiomatic model not a simple clone.
